WMF Framework/Router.swift

```diff
@@ -0,0 +1,165 @@
+@objc(WMFRouter)
+public class Router: NSObject {
+    public enum Destination: Equatable {
+        case inAppLink(_: URL)
+        case externalLink(_: URL)
+        case article(_: URL)
+        case articleHistory(_: URL, articleTitle: String)
+        case articleDiffCompare(_: URL, fromRevID: Int?, toRevID: Int?)
+        case articleDiffSingle(_: URL, fromRevID: Int?, toRevID: Int?)
+        case userTalk(_: URL)
+        case search(_: URL, term: String?)
+    }
+    
+    unowned let configuration: Configuration
+    required init(configuration: Configuration) {
+        self.configuration = configuration
+    }
+    
+    // From https://github.com/wikimedia/mediawiki-title
+    private let namespaceRegex = try! NSRegularExpression(pattern: "^(.+?)_*:_*(.*)$")
+    private let mobilediffRegexCompare = try! NSRegularExpression(pattern: "^mobilediff/([0-9]+)\\.\\.\\.([0-9]+)", options: .caseInsensitive)
+    private let mobilediffRegexSingle = try! NSRegularExpression(pattern: "^mobilediff/([0-9]+)", options: .caseInsensitive)
+    private let historyRegex = try! NSRegularExpression(pattern: "^history/(.*)", options: .caseInsensitive)
```

Comment:
This may be too broad, I'm trying to capture the article title in something like Special:History/Auguste_Rodin.